operator/pkg/apis/istio/v1alpha1/validation/validation.go

```diff
@@ -98,17 +98,18 @@ func deprecatedSettingsMessage(values *valuesv1alpha1.Values) string {
 }
 
 // validateFeatures check whether the config sematically make sense. For example, feature X and feature Y can't be enabled together.
-func validateFeatures(values *valuesv1alpha1.Values, _ *v1alpha1.IstioOperatorSpec) util.Errors {
+func validateFeatures(values *valuesv1alpha1.Values, iopSpec *v1alpha1.IstioOperatorSpec) util.Errors {
 	// When automatic mutual TLS is enabled, we check control plane security must also be enabled.
 	g := values.GetGlobal()
 	if g == nil {
 		return nil
 	}
-	m := g.GetMtls()
-	if m == nil {
-		return nil
+	mc := iopSpec.MeshConfig
+	val := true
+	if mc != nil && mc["enableAutoMtls"] != nil {
+		val = mc["enableAutoMtls"].(bool)
```

Comment:
can this be a non-bool and panic?

Reply:
I assumed we do a type validation first before that, but I can add it for safety